replacement_scan.go

```diff
@@ -0,0 +1,65 @@
+package duckdb
+
+/*
+   #include <stdlib.h>
+   #include <duckdb.h>
+   void replacement_scan_cb(duckdb_replacement_scan_info info, const char *table_name, void *data);
+   typedef const char cchar_t;
+*/
+import "C"
+import (
+	"sync"
+	"unsafe"
+)
+
+type ReplacementScanCallback func(tableName string) (string, []any, error)
+
+var replacementScanFnMxt sync.Mutex
+var replacementScanCallback ReplacementScanCallback
+
+func RegisterReplacementScan(connector *Connector, cb ReplacementScanCallback) error {
+	replacementScanFnMxt.Lock()
+	replacementScanCallback = cb
+	replacementScanFnMxt.Unlock()
+	C.duckdb_add_replacement_scan(connector.db, C.duckdb_replacement_callback_t(C.replacement_scan_cb), nil, C.duckdb_delete_callback_t(C.free))
+	return nil
```

Comment:
Do we need to return a `nil` error here? Is this for future extensibility?

Reply:
Yeah, let's remove it.